src/VisualStudio/Core/Def/Implementation/ProjectSystem/CPS/IProjectContext.cs

```diff
@@ -7,26 +7,34 @@
 namespace Microsoft.VisualStudio.LanguageServices.ProjectSystem
 {
     /// <summary>
-    /// Project context to initialize properties and items of a Workspace project created with <see cref="IProjectContextFactory.CreateProjectContext(string, string, string, Guid, string, Shell.Interop.IVsHierarchy, CommandLineArguments)"/>. 
+    /// Project context to initialize properties and items of a Workspace project created with <see cref="IProjectContextFactory.CreateProjectContext(string, string, string, Shell.Interop.IVsHierarchy)"/>. 
     /// </summary>
-    internal interface IProjectContext
+    internal interface IProjectContext : IDisposable
     {
         // Project properties.
-        ProjectId Id { get; }
         string DisplayName { get; set; }
         string ProjectFilePath { get; set; }
         Guid Guid { get; set; }
         string ProjectType { get; set; }
+        bool DesignTimeBuildStatus { get; set; }
 
         // Options.
         void SetCommandLineArguments(CommandLineArguments commandLineArguments);
+        void SetCommandLineArguments(string commandLineForOptions);
```

Comment:
I thought we were not sure on CPS side which one will be used? If we know we will use the new one for sure, I can remove the previous one (unless we see any future use for it).